### PR TITLE
bfloat16 support for probability distribution functions

### PR DESCRIPTION
**Pull request by rohithkrn**

### CODE REVIEW

File: aten/src/ATen/native/cuda/Distributions.cu [hunk 1]
```diff
@@ -404,8 +404,8 @@ Tensor& bernoulli_tensor_cuda_(Tensor &self, const Tensor& p_, Generator* gen_)
     rng_engine_inputs = gen->philox_engine_inputs(10);
   }
   auto p = std::get<0>(expand_inplace(self, p_.to(kCUDA)));
-  AT_DISPATCH_ALL_TYPES_AND2(
-    at::ScalarType::Half, at::ScalarType::Bool, self.scalar_type(), "bernoulli_tensor_cuda_self_", [&] {
+  AT_DISPATCH_ALL_TYPES_AND3(
+    at::ScalarType::Half, at::ScalarType::BFloat16, at::ScalarType::Bool, self.scalar_type(), "bernoulli_tensor_cuda_self_", [&] {
       using self_t = scalar_t;
       AT_DISPATCH_FLOATING_TYPES_AND_HALF(p.scalar_type(), "bernoulli_tensor_cuda_p_", [&] {
```
Comment (lcskrishna) [inline aten/src/ATen/native/cuda/Distributions.cu]: Can you check here if you require -> AT_DISPATCH_FLOATING_TYPES_AND2 here

Comment (rohithkrn) [reply aten/src/ATen/native/cuda/Distributions.cu]: You're right. I missed it.

**Comment by iotamudelta** [comment]
i guess the CI failures are real and should be addressed

**Comment by rohithkrn** [comment]
This PR depends on the macro defined in PR #505 and hence the failures in CI. I think it will work on CUDA too, but I will verify that. If not, should I dispatch kernels separately for HIPCC?

**Comment by iotamudelta** [comment]
In this branch: not necessary unless you want to get it done properly directly. We'd need to keep that in mind when upstreaming changes (but the full CI would also signal, of course).

**Comment by rohithkrn** [comment]
ROCm tests passed. Merging.

**Pull request merged**